concepts/05 UI Components/DataGrid/15 Columns/40 Customize Cells/3 Customize Row Apperance.md

```diff
@@ -0,0 +1,147 @@
+To customize the existing row layout, implement the [onRowPrepared](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onRowPrepared) event handler. This handler allows you to customize the existing row layout after it is generated. If a customization is not immediately applied, use the [repaintRows(rowIndexes)](/Documentation/ApiReference/UI_Components/dxDataGrid/Methods/#repaintRowsrowIndexes) method to repaint DataGrid rows.
```

Comment:
Again, maybe break this topic down into a couple of sections based on tasks: 

## Change Style Based on Row State (Focused, Selected, Hovered)

... all the CSS stuff goes here ... 

## Change Style Based on Row Data (Conditional Formatting) 

### Change Markup for the Entire Row 

... handle an event ... 

### Specify a Row Content Template

... talk about `dataRowTemplate` ...

Reply:
I updated both topics, however I didn't use titles of the inner (4th) level since they look awful on the website.